src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentitySource.cs

```diff
@@ -42,6 +42,12 @@ public enum ManagedIdentitySource
         /// <summary>
         /// The source to acquire token for managed identity is Service Fabric.
         /// </summary>
-        ServiceFabric
+        ServiceFabric,
+
+        /// <summary>
+        /// Indicates that the source is defaulted to IMDS since no environment variables are set.
+        /// This is used to detect the managed identity source.
+        /// </summary>
+        DefaultToImds
```

Comment:
```suggestion
        DefaultToImds
```
did we consider `FallbackToImds` or `ImdsAutoDetect` ?

Reply:
`DefaultToImds` sounds like a name that solves the purpose. I don't think the other names you mentioned suits here. Any reason you are suggesting a name change now? We discussed the names earlier too.